operator/controllers/instance_controller.go

```diff
@@ -0,0 +1,514 @@
+	}
+
+	// TODO: update the status with validated so the controller doesn't keep performing the
+	// validation on requeues.
+	if *instance.Spec.ValidateValues {
+		schemaJSON, err := yaml.YAMLToJSON([]byte(plan.Spec.Provisioning.Values.Schema))
+		if err != nil {
+			log.Error(err, "failed to validate values")
+			// TODO: update status with invalid values.
+			return ctrl.Result{}, nil
+		}
+		schema := gojsonschema.NewBytesLoader(schemaJSON)
+		valuesJSON, err := yaml.YAMLToJSON([]byte(instance.Spec.Values))
+		if err != nil {
+			log.Error(err, "failed to validate values")
+			// TODO: update status with invalid values.
+			return ctrl.Result{}, nil
+		}
+		values := gojsonschema.NewBytesLoader(valuesJSON)
+		result, err := gojsonschema.Validate(schema, values)
+		if err != nil {
+			log.Error(err, "failed to validate values")
+			// TODO: update status with invalid values.
+			return ctrl.Result{}, nil
+		}
+		if !result.Valid() {
+			log.Error(err, "failed to validate values")
+			// TODO: update status with invalid values. Including specific errors returned from
+			// result.Errors().
+			return ctrl.Result{}, nil
+		}
+	}
+
+	valuesSecretName := fmt.Sprintf("metabroker-%s-values", instance.Name)
+	if created, err := r.valuesSecret(ctx, instance, valuesSecretName, instance.Namespace); err != nil {
+		return ctrl.Result{}, err
+	} else if created {
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	return r.runProvisioningPod(ctx, ProvisioningRequest{releaseReq}, instance, plan, valuesSecretName)
+}
+
+// valuesSecret creates a Secret containing the values.yaml content if it doesn't exist yet.
+// The first return parameter represents whether the Secret was created or not.
+func (r *InstanceReconciler) valuesSecret(
+	ctx context.Context,
+	instance *servicebrokerv1alpha1.Instance,
+	valuesSecretName string,
+	namespace string,
+) (bool, error) {
+	namespacedName := types.NamespacedName{
+		Name:      valuesSecretName,
+		Namespace: namespace,
+	}
+	current := &corev1.Secret{}
+	if err := r.Get(ctx, namespacedName, current); err != nil {
+		if !errors.IsNotFound(err) {
+			return false, fmt.Errorf("failed to process values secret: %w", err)
+		}
+		desired := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      namespacedName.Name,
+				Namespace: namespacedName.Namespace,
+				// TODO: add proper labels.
+			},
+			Data: map[string][]byte{"values.yaml": []byte(instance.Spec.Values)},
+		}
+		if err := ctrl.SetControllerReference(instance, desired, r.Scheme); err != nil {
+			return false, fmt.Errorf("failed to process values secret: %w", err)
+		}
+		if err := r.Create(ctx, desired); err != nil {
+			return false, fmt.Errorf("failed to process values secret: %w", err)
+		}
+		return true, nil
+	}
+	return false, nil
+}
+
+// runProvisioningPod runs the provisioning pod. It creates the pod and its dependencies, ensuring
+// that upon successful completion, all the created resources are deleted.
```

Comment:
Hmm, this feels too concise. I'm thinking something like:
```suggestion
// runProvisioningPod ensures that provisioning takes place.  If the provisioning pod does not exist,
// that pod as well as any of its dependencies are created.  If the provisioning pod has already
// completed successfully, that pod as well as its dependencies are deleted.
```
Of course, that also points out
- It's not really a great function name.
- Something is missing to ensure this doesn't get in a provisioning loop (that is, re-creating the provisioning pod after it has been deleted).

Reply:
For the function name and comment, we can do better, but that doesn't really matter... it's just a wrapper function that's only called from the `Instance.Reconcile` method.

For the re-recreation of the pod, it would happen if there's an update to the `Instance` object. To fix this, I'll need to implement the object `Status`.